tests/msc4155/invite_filter_test.go

```diff
@@ -0,0 +1,181 @@
+		roomIDB := mustCreateRoomAndSync(t, evil_bob)
+		mustInviteRoomAndFail(t, evil_bob, roomIDB, alice.UserID)
+		mustHaveNoInviteInSyncResponse(t, alice)
+	})
+	t.Run("Can ignore a whole server", func(t *testing.T) {
+		mustSetInviteConfig(t, alice, InviteFilterConfig{
+			IgnoredServers: []string{hs2Name},
+		})
+		roomIDA := mustCreateRoomAndSync(t, bob)
+		bob.MustInviteRoom(t, roomIDA, alice.UserID)
+		roomIDB := mustCreateRoomAndSync(t, evil_bob)
+		evil_bob.MustInviteRoom(t, roomIDB, alice.UserID)
+		mustHaveNoInviteInSyncResponse(t, alice)
+	})
+	t.Run("Can glob serveral servers", func(t *testing.T) {
+		mustSetInviteConfig(t, alice, InviteFilterConfig{
+			BlockedServers: []string{"hs*"},
+		})
+		roomIDA := mustCreateRoomAndSync(t, bob)
+		mustInviteRoomAndFail(t, bob, roomIDA, alice.UserID)
+		roomIDB := mustCreateRoomAndSync(t, evil_bob)
+		mustInviteRoomAndFail(t, evil_bob, roomIDB, alice.UserID)
+		roomIDC := evil_alice.MustCreateRoom(t, map[string]interface{}{
+			"preset": "private_chat",
+		})
+		mustInviteRoomAndFail(t, evil_alice, roomIDC, alice.UserID)
+
+		mustHaveNoInviteInSyncResponse(t, alice)
+	})
+	t.Run("Can allow a user from a blocked server", func(t *testing.T) {
+		mustSetInviteConfig(t, alice, InviteFilterConfig{
+			AllowedUsers: []string{bob.UserID},
+			BlockedServers: []string{hs2Name},
+		})
+
+		roomIDBlocked := mustCreateRoomAndSync(t, evil_bob)
+		mustInviteRoomAndFail(t, evil_bob, roomIDBlocked, alice.UserID)
+		mustHaveNoInviteInSyncResponse(t, alice)
+		roomIDAllowed := mustCreateRoomAndSync(t, bob)
+		bob.MustInviteRoom(t, roomIDAllowed, alice.UserID)
+		alice.MustSyncUntil(t, client.SyncReq{}, client.SyncInvitedTo(alice.UserID, roomIDAllowed))
+		alice.MustJoinRoom(t, roomIDAllowed, []spec.ServerName{})
+		alice.MustSyncUntil(t, client.SyncReq{}, client.SyncJoinedTo(alice.UserID, roomIDAllowed))
+		bob.MustSyncUntil(t, client.SyncReq{}, client.SyncJoinedTo(alice.UserID, roomIDAllowed))
+	})
+	t.Run("Can block a user from an allowed server", func(t *testing.T) {
+		mustSetInviteConfig(t, alice, InviteFilterConfig{
+			BlockedUsers: []string{evil_bob.UserID},
+			AllowedServers: []string{hs2Name},
+		})
+
+		roomIDBlocked := mustCreateRoomAndSync(t, evil_bob)
+		mustInviteRoomAndFail(t, evil_bob, roomIDBlocked, alice.UserID)
+		mustHaveNoInviteInSyncResponse(t, alice)
+		roomIDAllowed := mustCreateRoomAndSync(t, bob)
+		bob.MustInviteRoom(t, roomIDAllowed, alice.UserID)
+		alice.MustSyncUntil(t, client.SyncReq{}, client.SyncInvitedTo(alice.UserID, roomIDAllowed))
+		alice.MustJoinRoom(t, roomIDAllowed, []spec.ServerName{})
+		alice.MustSyncUntil(t, client.SyncReq{}, client.SyncJoinedTo(alice.UserID, roomIDAllowed))
+		bob.MustSyncUntil(t, client.SyncReq{}, client.SyncJoinedTo(alice.UserID, roomIDAllowed))
+	})
+}
+
+// Tests that a given invite filter config is properly set
+func mustSetInviteConfig(t *testing.T, c *client.CSAPI, cfg InviteFilterConfig) {
+	b, err := json.Marshal(&cfg)
```

Comment:
Could you add a quick comment here explaining why the Marshal/Unmarshal dance is needed (to tranform `InviteFilterConfig` -> `map[string]interface{}` presumably).